cli/cmd/services_logs.go

```diff
@@ -0,0 +1,462 @@
+package cmd
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"sort"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/llamafarm/cli/cmd/utils"
+	"github.com/spf13/cobra"
+)
+
+// servicesLogsCmd displays logs for LlamaFarm services
+var servicesLogsCmd = &cobra.Command{
+	Use:   "logs",
+	Short: "View logs for LlamaFarm services",
+	Long: `Display logs from LlamaFarm services.
+
+Logs are stored in ~/.llamafarm/logs/<service>.log and can be viewed in real-time
+or as a static snapshot. When no service is specified, logs from all services are
+displayed with service name prefixes.
+
+Available services:
+  - server: The main FastAPI server
+  - rag: The RAG/Celery worker
+  - universal-runtime: The universal runtime server
+
+Examples:
+  lf services logs                           # Show logs from all services
+  lf services logs --tail 50                 # Show last 50 lines from all services
+  lf services logs --follow                  # Follow all service logs in real-time
+  lf services logs --service server          # Show all server logs
+  lf services logs -s rag --tail 50          # Show last 50 lines of RAG logs
+  lf services logs -s server --follow        # Follow server logs in real-time
+  lf services logs -s rag -f -n 100          # Follow RAG logs, starting with last 100 lines`,
+	Run: runServicesLogs,
+}
+
+func init() {
+	servicesCmd.AddCommand(servicesLogsCmd)
+
+	// Optional flag: which service to show logs for (if empty, show all)
+	servicesLogsCmd.Flags().StringP("service", "s", "", "Service to view logs for (server, rag, universal-runtime). If omitted, shows all services.")
+
+	// Optional flags
+	servicesLogsCmd.Flags().BoolP("follow", "f", false, "Follow log output (like tail -f)")
+	servicesLogsCmd.Flags().IntP("tail", "n", 0, "Number of lines to show from the end (0 = show all)")
+}
+
+// runServicesLogs is the main entry point for the services logs command
+func runServicesLogs(cmd *cobra.Command, args []string) {
+	serviceName, _ := cmd.Flags().GetString("service")
+	follow, _ := cmd.Flags().GetBool("follow")
+	tailLines, _ := cmd.Flags().GetInt("tail")
```

Comment:
Maybe we should default this to 100 or something. I feel like it's pretty common to limit the default so that we don't end up potentially dumping MBs of logs to the console.